provider/oracle/constraints.go

```diff
@@ -0,0 +1,89 @@
+package oracle
+
+import "github.com/juju/juju/constraints"
+
+// cons provides a wrapper-adaptor arount the default constraints
```

Comment:
why is this needed? why can you not just use the existing constraints.Validator, and register which constraints are unsupported, etc.?